opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py

```diff
@@ -133,6 +162,8 @@ class ProbabilitySampler(Sampler):
     """
 
     def __init__(self, rate: float):
+        if rate < 0.0 or rate > 1.0:
+            raise ValueError("Probability must be in range [0.0, 1.0].")
```

Comment:
Github won't let me comment on the L182 directly, but do we need that setter for rate? Seems like that shouldn't be part of the public api

Reply:
I think the purpose of the setter for rate is to change `_rate` and `_bound` in tangent, since `rate` is a property of the class, I am assuming it was meant to be "settable" as well. I am 50/50 in letting this be part of the public api though.